ansible_collections/arista/avd/roles/eos_designs/python_modules/ip_addressing/utils.py

```diff
@@ -107,3 +109,40 @@ def _mlag_odd_id_based_offset(self: "AvdIpAddressing") -> int:
             odd_id = self._mlag_secondary_id
 
         return int((odd_id - 1) / 2)
+
+    def _get_parallel_uplink_index(self: "AvdIpAddressing", uplink_switch_index: int) -> int:
+        uplink_switch = self.shared_utils.uplink_switches[uplink_switch_index]
+        uplink_switch_indexes = [index for index, value in enumerate(self.shared_utils.uplink_switches) if value == uplink_switch]
+        # Find index of uplink_interface going to the same uplink_switch (in case of parallel uplinks)
+        return uplink_switch_indexes.index(uplink_switch_index)
+
+    def _get_downlink_ipv4_pool(self: "AvdIpAddressing", uplink_switch_index: int) -> str | None:
+        uplink_switch_interface = self.shared_utils.uplink_switch_interfaces[uplink_switch_index]
+        uplink_switch = self.shared_utils.uplink_switches[uplink_switch_index]
+        peer_facts = self.shared_utils.get_peer_facts(uplink_switch, required=True)
+        downlink_pools = get(peer_facts, "downlink_pools")
+
+        if not downlink_pools:
+            return None
+
+        for downlink_pool_and_interfaces in downlink_pools:
+            downlink_interfaces = range_expand(get(downlink_pool_and_interfaces, "downlink_interfaces"))
+
+            if uplink_switch_interface in range_expand(downlink_interfaces):
```

Comment:
range_expand will expand a list of ranges, so I don't think you need to expand again.
```suggestion
            if uplink_switch_interface in downlink_interfaces:
```